### PR TITLE
[1LP][RFR] fixes for test_vm_reconfigure

### PR DESCRIPTION
**Pull request by lkhomenk**
Purpose or Intent
=================

__Fixing__ test_vm_reconfig_add_remove_disk_cold because it is currently doesn't remove disks that were added to VM from the env
+ Resolving few TODOs

{{pytest: --long-running --use-provider vsphere65-nested --use-provider rhv41 -k test_vm_reconfig_add_remove_disk_cold cfme/tests/infrastructure/test_vm_reconfigure.py}}

### CODE REVIEW

File: widgetastic_manageiq/vm_reconfigure.py [hunk 1]
```diff
@@ -28,11 +28,13 @@ class DisksTable(VanillaTable):
...
+        'ControllerType': BootstrapSelect(id='Controller'),
         # Workaround necessary until BZ 1524960 is resolved
-        4: BootstrapSelect(id='hdUnit'),
-        3: BootstrapSelect(id='hdUnit'),
-        'Dependent': BootstrapSwitch(
-            name=VersionPick({Version.lowest(): 'cb_dependent', '5.9': 'vm.cb_dependent'})),
+        5: BootstrapSelect(id='hdUnit'),  # for VMware in 5.9
+        4: BootstrapSelect(id='hdUnit'),  # for VMware in 5.8
+        3: BootstrapSelect(id='hdUnit'),  # for RHEVM in 5.9
+        'Dependent': VersionPick({Version.lowest(): BootstrapSwitch(name='cb_dependent'),
```
Comment (lkhomenk) [inline widgetastic_manageiq/vm_reconfigure.py]: workaround, name/id can't be version_picked atm - `quote` fails on VersionPick object. and we can't pass `locator` to  `BootstrapSwitch`, which could cover  both cases. another PR needed for that

Comment (mfalesni) [reply widgetastic_manageiq/vm_reconfigure.py]: True, versionpick does not work in second level nesting which widgets kwarg is

Comment (lkhomenk) [reply widgetastic_manageiq/vm_reconfigure.py]: `name` version pick doesn't work here as it's a Table with widgets and it's parameter - second level.

File: cfme/tests/infrastructure/test_vm_reconfigure.py [hunk 2]
```diff
@@ -81,21 +81,20 @@ def test_vm_reconfig_add_remove_hw_cold(
...
-        lambda: small_vm.configuration == orig_config, timeout=360, delay=45,
-        fail_func=small_vm.refresh_relationships,
-        message="confirm that previously-added disk was removed")
+        size=500, size_unit='MB', type=disk_type, mode=disk_mode)
+
+    add_disk_request = small_vm.reconfigure(new_config)
+    wait_for(add_disk_request.is_succeeded, timeout=360, delay=45,
+             fail_func=small_vm.refresh_relationships,
+             message="confirm that disk was added")
+    soft_assert(lambda: small_vm.configuration == new_config, "Disk wasn't added to VM config")
```
Comment (quarckster) [inline cfme/tests/infrastructure/test_vm_reconfigure.py]: **REQUIRED**
I don't think `soft_assert` makes sense here. If a disk is not added, there will nothing to remove further.

Comment (lkhomenk) [reply cfme/tests/infrastructure/test_vm_reconfigure.py]: true.

**Comment by lkhomenk** [comment]
works perfect with  https://github.com/RedHatQE/widgetastic.core/pull/80

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by lkhomenk** [comment]
@quarckster please recheck it

File: cfme/tests/infrastructure/test_vm_reconfigure.py [hunk 3]
```diff
@@ -86,19 +86,30 @@ def test_vm_reconfig_add_remove_disk_cold(
...
+    # Add disk UI verification
     wait_for(
-        lambda: small_vm.configuration == new_config, timeout=360, delay=45,
+        lambda: small_vm.configuration.num_disks == new_config.num_disks, timeout=360, delay=45,
         fail_func=small_vm.refresh_relationships,
         message="confirm that disk was added")
-
-    small_vm.reconfigure(orig_config)
+    assert small_vm.configuration.num_disks == new_config.num_disks,\
+        "Disk wasn't added to VM config"
```
Comment (quarckster) [inline cfme/tests/infrastructure/test_vm_reconfigure.py]: **REQUIRED**
```python
msg = "Disk wasn't added to VM config"
assert small_vm.configuration.num_disks == new_config.num_disks, msg
```

File: cfme/tests/infrastructure/test_vm_reconfigure.py [hunk 4]
```diff
@@ -86,19 +86,30 @@ def test_vm_reconfig_add_remove_disk_cold(
...
+    wait_for(remove_disk_request.is_succeeded, timeout=360, delay=45,
+             message="confirm that previously-added disk was removed")
+    # Remove disk UI verification
     wait_for(
-        lambda: small_vm.configuration == orig_config, timeout=360, delay=45,
+        lambda: small_vm.configuration.num_disks == orig_config.num_disks, timeout=360, delay=45,
         fail_func=small_vm.refresh_relationships,
         message="confirm that previously-added disk was removed")
+    assert small_vm.configuration.num_disks == orig_config.num_disks,\
+        "Disk wasn't removed from VM config"
```
Comment (quarckster) [inline cfme/tests/infrastructure/test_vm_reconfigure.py]: **REQUIRED**
```python
msg = "Disk wasn't removed from VM config"
assert small_vm.configuration.num_disks == orig_config.num_disks, msg
```

File: cfme/infrastructure/virtual_machines.py [hunk 5]
```diff
@@ -884,33 +885,40 @@ def reconfigure(self, new_configuration=None, changes=None, cancel=False):
                 row.type.fill(disk.type)
                 # Unit first, then size (otherwise JS would try to recalculate the size...)
                 if self.provider.one_of(RHEVMProvider):
-                    # Workaround necessary until BZ 1524960 is resolved
+                    # TODO: Workaround necessary until BZ 1524960 is resolved
+                    # -- block start --
```
Comment (mshriver) [inline cfme/infrastructure/virtual_machines.py]: :+1:

**Pull request merged**